src/risk_distributions/risk_distributions.py

```diff
@@ -469,53 +469,19 @@ def get_parameters(cls, weights: Parameters,
 
         return weights, params
 
-    def pdf(self, x: Union[pd.Series, np.ndarray, float, int]) -> Union[pd.Series, np.ndarray, float]:
-        single_val = isinstance(x, (float, int))
-        values_only = isinstance(x, np.ndarray)
-
-        x, weights = format_call_data(x, self.weights)
-
-        computable = weights[(weights.sum(axis=1) != 0) & ~np.isnan(x)].index
-
-        p = pd.Series(np.nan, index=x.index)
-
-        if not computable.empty:
-            p.loc[computable] = 0
-            for name, parameters in self.parameters.items():
-                w = weights.loc[computable, name]
-                params = parameters.loc[computable] if len(parameters) > 1 else parameters
-                p += w * self._distribution_map[name](parameters=params).pdf(x.loc[computable])
-
-        if single_val:
-            p = p.iloc[0]
-        if values_only:
-            p = p.values
-        return p
 
-    def ppf(self, q: Union[pd.Series, np.ndarray, float, int]) -> Union[pd.Series, np.ndarray, float]:
-        single_val = isinstance(q, (float, int))
-        values_only = isinstance(q, np.ndarray)
-
-        q, weights = format_call_data(q, self.weights)
+    def pdf(self, x: Union[pd.Series, np.ndarray, float, int]) -> Union[pd.Series, np.ndarray, float]:
```

Comment:
The pdf and cdf function implementation shouldn't change.  It is still a weighted sum of the component distributions.